catalog/registry/catalog-registry-api-impl/src/main/java/ddf/catalog/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -0,0 +1,230 @@
+        return map;
+    }
+
+    @Override
+    public boolean isPushAllowed() {
+        return pushAllowed;
+    }
+
+    @Override
+    public boolean isPullAllowed() {
+        return pullAllowed;
+    }
+
+    @Override
+    public CreateResponse create(CreateRequest request) throws IngestException {
+        if (!pushAllowed) {
+            throw new IngestException("Push (write) operations are not allowed for this registry");
+        }
+        return super.create(request);
+    }
+
+    @Override
+    public UpdateResponse update(UpdateRequest request) throws IngestException {
+
+        if (!pushAllowed) {
+            throw new IngestException("Push (write) operations are not allowed for this registry");
+        }
+
+        Map<String, Metacard> updatedMetacards = new HashMap<>();
+        Map<String, Metacard> origMetacards = new HashMap<>();
+
+        for (Map.Entry<Serializable, Metacard> entry : request.getUpdates()) {
+
+            updatedMetacards.put(getRegistryId(entry.getValue()), entry.getValue());
+        }
+
+        OperationTransaction opTrans =
+                (OperationTransaction) request.getPropertyValue(Constants.OPERATION_TRANSACTION_KEY);
+
+        for (Metacard mcard : opTrans.getPreviousStateMetacards()) {
+            origMetacards.put(getRegistryId(mcard), mcard);
+        }
+
+        //update the new metacards with the id from the orig so that they can be found on the remote system
+        try {
+            for (Map.Entry<String, Metacard> entry : updatedMetacards.entrySet()) {
+                setMetacardExtID(entry.getValue(),
+                        origMetacards.get(entry.getKey())
+                                .getId());
+            }
+        } catch (ParserException e) {
+            throw new IngestException("Could not update metacards id", e);
+        }
+
+        return super.update(request);
+    }
+
+    @Override
+    public DeleteResponse delete(DeleteRequest request) throws IngestException {
+        if (!pushAllowed) {
+            throw new IngestException("Push (write) operations are not allowed for this registry");
+        }
+        return super.delete(request);
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
+        if (!pullAllowed) {
+            throw new UnsupportedQueryException(
+                    "Pull (read) operations are not allowed for this registry");
+        }
+        return super.query(request);
+    }
+
+    public void setPushAllowed(boolean pushAllowed) {
+        this.pushAllowed = pushAllowed;
+    }
+
+    public void setPullAllowed(boolean pullAllowed) {
+        this.pullAllowed = pullAllowed;
+    }
+
+    private String getRegistryId(Metacard mcard) {
+        Attribute attr = mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+        if (attr != null && attr.getValue() instanceof String) {
```

Comment:
I read this one too quickly and saw it the same as the one I'd commented on earlier. You're correct; it needs to be checked.